### ISSUE TITLE
Meta.ddl() generates broken DDL for DB2 and Derby table aliases

### ISSUE THREAD

**Issue by mgramin** [jOOQ/jOOQ#15325]
### Expected behavior

We expect that Meta.ddl() exports a valid DDL for DB2 aliases:

```sql
CREATE TABLE DB2INST1.TEST_TABLE (
	ID VARCHAR(50)
);
CREATE ALIAS t FOR test_table;
```

### Actual behavior

Meta.ddl() exports "create table" DDL for aliases (instead of "create alias"):
```sql
create table "DB2INST1"."T" (
  "ID" varchar(50)
)
create table "DB2INST1"."TEST_TABLE" (
  "ID" varchar(50)
)
```

### Steps to reproduce the problem

Create a table and alias:
```sql
CREATE TABLE DB2INST1.TEST_TABLE (
	ID VARCHAR(50)
);
CREATE ALIAS t FOR test_table;
```

Run this code:
```java
        Connection conn = DriverManager.getConnection("jdbc:db2://localhost:50000/sample", "db2inst1", "secret_password");
        Configuration configuration = new DefaultConfiguration().set(conn).set(SQLDialect.DB2);
        Meta meta = using(configuration).meta();

        Arrays.stream(meta.filterSchemas(v -> v.getName().equalsIgnoreCase("DB2INST1"))
                        .ddl()
                        .queries())
                .map(Object::toString)
                .forEach(System.out::println);
```

Executing the resulting query against a different database or schema:
```sql
create table "DB2INST1"."T" (
  "ID" varchar(50)
)
create table "DB2INST1"."TEST_TABLE" (
  "ID" varchar(50)
)
```

As a result, we have 2 tables, instead of one table and one alias.

### jOOQ Version

jOOQ Professional Edition 3.18.4

### Database product and version

DB2 v11.5.8.0

### Java Version

openjdk 17.0.2 2022-01-18

### OS Version

Microsoft Windows [Version 10.0.19044.2846]

### JDBC driver name and version (include name if unofficial driver)

com.ibm.db2.jcc:db2jcc:db2jcc4

**Comment by lukaseder** [comment]
Thanks for your report. While supporting aliases (or "synonyms", see: https://github.com/jOOQ/jOOQ/issues/9574) is a big task thoughout jOOQ, I can definitely see how the generated DDL can be problematic. Though, we can't just omit the aliases from `Meta` output because generating DDL isn't the only purpose of the `Meta` API. In other cases, the alias should indeed behave like the table itself, including listing all of its columns.

Apart from the DDL being clearly wrong, what are your expectations from a fix? Would anything specific, short of implementing all of #9574 be helpful?

**Comment by mgramin** [comment]
Thank you for the clarification.

I think it would be very useful to have the option to ignore aliases from a list of tables.

**Comment by lukaseder** [comment]
Had a closer look now. We do resolve some synonyms in Oracle's code generator. The reason being that synonyms can be referenced by PL/SQL packages (parameter or return types). We have to resolve them to look up the *actual* type, e.g. a UDT, in order to generate correct code. This was done here, in jOOQ 3.5:

- https://github.com/jOOQ/jOOQ/issues/3552

I guess that the default behaviour in jOOQ should always be to resolve synonyms and pretend they don't exist. In a second step, #9574 can be implemented, and code generation as well as runtime support for synonyms can be added, but again, that's a much bigger task. In the short run we'll:

- Resolve all synonym references
- Remove all synonym declarations (which is what this issue here is about)

The resolution step isn't really optional. If you have synonyms, then there's risk of them being used in views, for example. Resolution will be a prerequisite for the proper implementation of a previous issue of yours:

- https://github.com/jOOQ/jOOQ/issues/15327

I have created a new feature request to expedite parts of the `Synonym` support task:

- https://github.com/jOOQ/jOOQ/issues/15335

So, in short, I don't think it's worth adding flags for such specific things like ignoring them from DDL export. We need a more long term strategy, and I'll implement that, instead.

**Comment by lukaseder** [comment]
I'm looking into this now for jOOQ 3.20:

- https://github.com/jOOQ/jOOQ/issues/9574

**Comment by lukaseder** [comment]
OK, after quite a bit of effort in #9574, I realised that this issue was mainly about omitting fetching the aliases in `org.jooq.Meta`, not making sure aliases are exported correctly via `ddl()`... That fix would have been very simple to implement already at the time. Sorry for the delay. I'll implement this right away now.

**Comment by lukaseder** [comment]
In fact, this is the same as https://github.com/jOOQ/jOOQ/issues/2448, for Oracle.

**Comment by lukaseder** [comment]
H2 is also affected.

I suspect that rather than implementing dialect-specific inclusive filters of table types, such as:

```java
types = new String[] { "SYSTEM TABLE", "TABLE", "VIEW" };
```

It might be better to filter out all aliases and synonyms for all dialects, because we never want to include those. I'll improve the fix per this issue.
